api/api/test/test_validator.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+# Copyright (C) 2015-2019, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import pytest
+
+from api import validator
+
+
+@pytest.mark.parametrize('exp, regex_name', [
+    ('54355', 'numbers'),
+    ('43,21,34', 'array_numbers'),
+    ('file-test_name1', 'names'),
+    ('file_1,file_2,file-3', 'array_names'),
+    ('/var/ossec/etc/internal_options', 'paths'),
+    ('/var/ossec/etc/rules/local_rules.xml', 'paths'),
+    ('20190226', 'dates'),
+    ('192.168.122.255', 'ips'),
+    ('any', 'ips'),
+    ('alphanumeric1_param2', 'alphanumeric_param'),
+    ('sort param-', 'sort_param'),
+    ('search param3', 'search_param'),
+    ('select_param2', 'select_param'),
+    ('5-35', 'ranges'),
+    ('e4d909c290d0fb1ca068ffaddf22cbd0', 'hashes'),
+    ('449e3b6ffd9b484c5c645321edd4d610', 'ossec_key'),
+    ('1d', 'timeframe_type'),
+    ('12h', 'timeframe_type'),
+    ('40m', 'timeframe_type'),
+    ('60s', 'timeframe_type'),
+    ('', 'empty_boolean'),
+    ('true', 'empty_boolean'),
+    ('false', 'empty_boolean'),
+    ('yes', 'yes_no_boolean'),
+    ('no', 'yes_no_boolean'),
+    ('true', 'boolean'),
+    ('false', 'boolean'),
+    ('param1 param2 param3', 'query_param'),
+    ('xml', 'type_format'),
+    ('json', 'type_format'),
+    ('etc/ossec.conf', 'relative_paths'),
+    ('etc/rules/new_rules2.xml', 'relative_paths'),
+    ('etc/lists/new_lists3', 'relative_paths')
+])
+def test_validation_check_exp_ok(exp, regex_name):
+    assert validator.check_exp(exp, regex_name) is True
+
+
+@pytest.mark.parametrize('exp, regex_name', [
+    ('543a', 'numbers'),
+    ('43a,21,34', 'array_numbers'),
+    ('file-$', 'names'),
+    ('file_1$,file_2#,file-3', 'array_names'),
+    ('/var/ossec/etc/internal_options$', 'paths'),
+    ('/var/ossec/etc/rules/local_rules.xml()', 'paths'),
+    ('2019-02-26', 'dates'),
+    ('192.168.122.256', 'ips'),
+    ('192.266.1.1', 'ips'),
+    ('alphanumeric1_$param2', 'alphanumeric_param'),
+    ('sort param@', 'sort_param'),
+    ('search param;', 'search_param'),
+    ('select_param2;', 'select_param'),
+    ('5-35-32', 'ranges'),
+    ('$$d909c290d0fb1ca068ffaddf22cbd0', 'hashes'),
+    ('449e3b6ffd9b484c5c645321edd4d61$', 'ossec_key'),
+    ('1j', 'timeframe_type'),
+    ('12x', 'timeframe_type'),
+    ('yes', 'empty_boolean'),
+    ('truee', 'empty_boolean'),
+    ('true', 'yes_no_boolean'),
+    ('false', 'yes_no_boolean'),
+    ('correct', 'boolean'),
+    ('wrong', 'boolean'),
+    ('param1,param2,param3', 'query_param'),
+    ('txt', 'type_format'),
+    ('exe', 'type_format'),
+    ('etc/internal_options', 'relative_paths'),
+    ('../../path', 'relative_paths'),
+    ('/var/ossec/etc/lists/new_lists3', 'relative_paths')
+])
+def test_validation_check_exp_ko(exp, regex_name):
+    assert validator.check_exp(exp, regex_name) is False
```

Comment:
I would prefer assert(not validator.check_exp(exp, regex_name))

Reply:
Done!